From c937ca3cf6577edc2025039f094c76f285e98615 Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbin.du@gmail.com>
Date: Mon, 22 Dec 2014 19:39:54 +0800
Subject: [PATCH 1/1] usb/f_mtp: return error code if tranfer error in
 receive_file_work function

receive_file_work() function should report error if it detect the urb
is not successfully transfered.

Tracked_On: https://jira01.devtools.intel.com/browse/IMINAN-9061
Signed-off-by: Du, Changbin <changbin.du@gmail.com>

Change-Id: Ia5e610f79296f3d76f6f64c911c668f293ce661f
---
 drivers/usb/gadget/f_mtp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/f_mtp.c b/drivers/usb/gadget/f_mtp.c
index 710337a189b8..92f9880c32f8 100644
--- a/drivers/usb/gadget/f_mtp.c
+++ b/drivers/usb/gadget/f_mtp.c
@@ -864,6 +864,10 @@ static void receive_file_work(struct work_struct *data)
 					usb_ep_dequeue(dev->ep_out, read_req);
 				break;
 			}
+			if (read_req->status) {
+				r = read_req->status;
+				break;
+			}
 			/* if xfer_file_length is 0xFFFFFFFF, then we read until
 			 * we get a zero length packet
 			 */
-- 
1.9.1

